core/src/main/scala/chisel3/experimental/hierarchy/core/Instance.scala

```diff
@@ -107,9 +108,33 @@ object Instance extends SourceInfoDoc {
     implicit sourceInfo: SourceInfo,
     compileOptions:      CompileOptions
   ): Instance[T] = {
+    // Check to see if the module is already defined
+    val existingMod = Builder.components.map {
+      case c: DefModule if c.id == definition.proto => Some(c)
+      case c: DefBlackBox if c.name == definition.proto.name => Some(c)
+      case _ => None
+    }.flatten
+
+    if (existingMod.isEmpty) {
+      // Add a module that will get extracted later on so that FIRRTL does not
```

Comment:
what does "extracted" mean? Do you mean:
```suggestion
      // Add a Definition that will get emitted as an ExtModule so that FIRRTL does not
```